frontend/src/components/FragmentComment.js

```diff
@@ -0,0 +1,50 @@
+import '../styles/FragmentComment.css';
+import React from 'react';
+
+function FragmentComment({ children }) {
+  try {
+    const citationRegex = /^\[.*?\]\s*\n(.*)$/m;
+    if (citationRegex.test(children)) {
+      let [citation, commentaire] = children[0].split(/\n/);
+      citation = citation.replace(/[\[\]]/g, '');
+      return <p
+        onMouseEnter={
+          e => {
+            e.preventDefault();
+            e.stopPropagation();
+            highlight(citation);
+          }
+        }
+        onMouseLeave={
+          e => {
+            e.preventDefault();
+            e.stopPropagation();
+            unHighlight();
+          }
+        }
+        className="fragmentComment"
+        title="Highlight in document"
+      >{commentaire}</p>;
+    }
+  } catch (e) {}
+
+  function highlight(texte) {
+    let passages = document.getElementsByClassName('main col');
+    for (let passage of passages) {
+      if (passage.innerHTML.includes(texte)) {
+        passage.innerHTML = passage.innerHTML.replace(texte, '<highlighted>' + texte + '</highlighted>');
```

Comment:
`texte`: Please name variables with english term.

Reply:
Please note that you could use `[Array].forEach` for a more modern syntax.

Reply:
Thanks for the advice, this is now fixed in the last commit.